plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -985,6 +986,7 @@ public void rollback()
         TupleDomain<IcebergColumnHandle> newUnenforcedConstraint = constraint.getSummary()
```

Comment:
this causes the "remaining" filter to exclude the structural predicates, so the engine thinks that it was subsumed by the connector, and will remove from the filter node. so I think this will produce wrong results.

we should add a test which applies a conjunction of complex and primitive typed filters, where the former is more selective than the latter.  (e.g. id and struct_t, but two rows with different struct_t share the same id)

Reply:
Thanks. You're right, that test case exposed a problem. Instead I split out a separate remainingConstrant from unenforcedConstraint which has all non-partition domains.